pkg/util/http/client.go

```diff
@@ -31,86 +27,68 @@ func SanitizeURL(message string) string {
 	return apiKeyRegExp.ReplaceAllString(message, apiKeyReplacement)
 }
 
-// CreateHTTPTransport creates an *http.Transport for use in the agent
-func CreateHTTPTransport() *http.Transport {
-	tlsConfig := &tls.Config{
-		InsecureSkipVerify: config.Datadog.GetBool("skip_ssl_validation"),
+// Client wraps (http.Client).Do and resets the underlying connections at the
+// configured interval
+type Client struct {
+	httpClient    *http.Client
+	newHTTPClient func() *http.Client
+	resetInterval time.Duration
+
+	lastReset time.Time
+	mutex     sync.RWMutex
```

Comment:
The idiom in Go is to call it `mu` and have it grouped together with the fields it protects. Something [like here](https://github.com/golang/go/blob/a38a917aee626a9b9d5ce2b93964f586bf759ea0/src/log/syslog/syslog.go#L83):
```go
type ResetClient struct {
    httpClient.   *http.Client
    newHTTPClient func() *http.Client

    mu            sync.RWMutex // guards below fields
    lastReset     time.Duration
    resetInterval time.Duration
}
```

Reply:
Because you're also using it for the read/write of `httpClient` then, all fields are guarded, you could go with composition as well:
```
type ResetClient struct {
    sync.RWMutex

    httpClient    *http.Client
    newHTTPClient func() *http.Client

    lastReset     time.Duration
    resetInterval time.Duration

}
```

Reply:
addressed. With current code, all the fields actually aren't guarded (`c.resetInterval` isn't guarded because it's a "config" param that should never be written to after the struct initialization)

Reply:
Embedding the mutex, even though many times convenient makes it exported and its methods become part of `ResetClient`. I think it's best unexported to avoid dodgy use.